### PR TITLE
[prometheus-kafka-exporter] fix Chart.yaml - remove engine: gotpl

### PR DESCRIPTION
**Pull request by arukiidou**
### What this PR does / why we need it

- fix Chart.yaml - remove engine: gotpl , see #2813
  - drop Helm v2 support
  - add Chart.lock

#### Which issue this PR fixes

- https://github.com/prometheus-community/helm-charts/issues/2813

#### Special notes for your reviewer

- This does not change the template. 
  - But, I think it's a major bump, because dropped Helmv2

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
  - [x] Has breaking change, bump major.
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

**Comment by arukiidou** [comment]
@gkarthiks please take a look

**Comment by zeritti** [comment]
@arukiidou Please, resolve the conflicts.

File: charts/prometheus-kafka-exporter/README.md [hunk 1]
```diff
@@ -4,29 +4,27 @@ A Prometheus exporter for [Apacher Kafka](https://kafka.apache.org/) metrics.
...
 ## Prerequisites
 
-- Kubernetes 1.8+ with Beta APIs enabled
+- Kubernetes 1.19+
+- Helm 3
+
+Helm v2 was no longer supported from chart version 2.0.0.
 
-## Get Repo Info
+## Get repository Info
```
Comment (zeritti) [inline charts/prometheus-kafka-exporter/README.md:14]: Will you please, change "repository" to "Repository" as is the case in other headings?

File: charts/prometheus-kafka-exporter/README.md [hunk 2]
```diff
@@ -4,29 +4,27 @@ A Prometheus exporter for [Apacher Kafka](https://kafka.apache.org/) metrics.
...
 This chart bootstraps a [Kafka Exporter](https://github.com/danielqsj/kafka_exporter) deployment on a [Kubernetes](http://kubernetes.io) cluster using the [Helm](https://helm.sh) package manager.
 
-
 ## Prerequisites
 
-- Kubernetes 1.8+ with Beta APIs enabled
+- Kubernetes 1.19+
+- Helm 3
+
+Helm v2 was no longer supported from chart version 2.0.0.
```
Comment (zeritti) [inline charts/prometheus-kafka-exporter/README.md:12]: As this is a statement about all the coming chart's releases, "is" would be more appropriate than "was", would it not?

Comment (arukiidou) [reply charts/prometheus-kafka-exporter/README.md]: fixed.  https://github.com/prometheus-community/helm-charts/pull/3199/commits/a4311e1b542a60ca1cc1364b2f2e295d97d3b82e

File: charts/prometheus-kafka-exporter/README.md (hunk 1 above)
Comment (arukiidou) [reply charts/prometheus-kafka-exporter/README.md]: fixed.  https://github.com/prometheus-community/helm-charts/pull/3199/commits/e5c13e9aa397ea2d97025c0748d978d24cc26117

**Comment by arukiidou** [comment]
> @arukiidou Please, resolve the conflicts.

rebased it

**Review by zeritti — APPROVED** [review]

**Comment by zeritti** [comment]
Thank you, @arukiidou, for your contribution.

**Pull request merged**